pghoard/basebackup.py

```diff
@@ -210,8 +210,15 @@ def run_piped_basebackup(self):
                       command, self.pid, compressed_basebackup)
 
         stream_target = os.path.join(temp_basebackup_dir, "data.tmp")
-        original_input_size, compressed_file_size, metadata = \
-            self.basebackup_compression_pipe(proc, stream_target)
+
+        # catch any os level exceptions such out of disk space, so that the underlying
+        # OS process gets properly cleaned up by check_command_success
+        try:
+            original_input_size, compressed_file_size, metadata = \
+                self.basebackup_compression_pipe(proc, stream_target)
+        except OSError as e:
+            self.log.error(str(e))
```

Comment:
This log message could be improved a bit; knowing where this happened (solely from log line) is often needed. So something like

```
self.log.error("basebackup_compression_pipe(%r, %r) failed with %r. Ignoring; check_command_success will detect this.", proc.stream_target, e)
```

would be more useful than just the contents of the exception (which is just a bit generic "out of disk space").

Reply:
I applied your suggested log line verbatim thanks for the suggestion